src/Schedule.h

```diff
@@ -94,26 +111,35 @@ enum class PrefetchBoundStrategy {
  * function is done by generating a loop nest that spans its
  * dimensions. We schedule the inputs to that function by
  * recursively injecting realizations for them at particular sites
- * in this loop nest. A LoopLevel identifies such a site. */
+ * in this loop nest. A LoopLevel identifies such a site. The site
+ * can either be a loop nest within all stages of a function
+ * or it can refer to a loop nest within a particular function's
+ * stage (initial definition or updates).
+ */
 class LoopLevel {
     template <typename T> friend class ScheduleParam;
     friend class ::Halide::Internal::ScheduleParamBase;
 
     Internal::IntrusivePtr<Internal::LoopLevelContents> contents;
 
     explicit LoopLevel(Internal::IntrusivePtr<Internal::LoopLevelContents> c) : contents(c) {}
-    EXPORT LoopLevel(const std::string &func_name, const std::string &var_name, bool is_rvar);
+    EXPORT LoopLevel(const std::string &func_name, const std::string &var_name,
+                     bool is_rvar, int stage_index);
 
     /** Mutate our contents to match the contents of 'other'. This is a potentially
      * dangerous operation to do if you aren't careful, and exists solely to make
      * ScheduleParam<LoopLevel> easy to implement; hence its private status. */
     EXPORT void copy_from(const LoopLevel &other);
 
 public:
+    /** Return the index of the function stage associated with this loop level.
+     * Asserts if undefined */
+    EXPORT int stage_index() const;
+
     /** Identify the loop nest corresponding to some dimension of some function */
     // @{
-    EXPORT LoopLevel(Internal::Function f, VarOrRVar v);
-    EXPORT LoopLevel(Func f, VarOrRVar v);
+    EXPORT LoopLevel(Internal::Function f, VarOrRVar v, int stage_level = -1);
+    EXPORT LoopLevel(Func f, VarOrRVar v, int stage_level = -1);
```

Comment:
This is a user-facing ctor: should a user ever pass anything other than -1? What happens if they do?